src/_pytest/pathlib.py

```diff
@@ -795,6 +787,35 @@ def resolve_pkg_root_and_module_name(
     raise CouldNotResolvePathError(f"Could not resolve for {path}")
 
 
+def _is_namespace_package(module_path: Path) -> bool:
+    # If the path has na __init__.py file, it means it is not
+    # a namespace package:.
+    # https://packaging.python.org/en/latest/guides/packaging-namespace-packages.
+    if (module_path / "__init__.py").is_file():
+        return False
+
+    module_name = module_path.name
+
+    # Empty module names break find_spec.
+    if not module_name:
+        return False
+
+    # Modules starting with "." indicate relative imports and break find_spec, and we are only attempting
+    # to find top-level namespace packages anyway.
+    if module_name.startswith("."):
```

Comment:
What if the dir contains a `.` not at the beginning e.g. `dir/py.test`?

Reply:
We can drop this check and catch an error from `find_spec` instead. 👍